### PR TITLE
[fix](function) MicroSecondsSub without scale 

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes
Added the computeSignature function for millisecond/microsecond calculation functions to generate parameters and return values with the appropriate precision.
Modified the microSecondsAdd function, which was used for constant folding, because constant folding uses the precision of the parameters for calculation. However, for millisecond/microsecond calculations, it is necessary to set the precision to the maximum to ensure correct display.

before
```
mysql> SELECT MICROSECONDS_SUB('2010-11-30 23:50:50', 2);
+-------------------------------------------------------------------+
| microseconds_sub(cast('2010-11-30 23:50:50' as DATETIMEV2(0)), 2) |
+-------------------------------------------------------------------+
| 2010-11-30 23:50:49                                               |
+-------------------------------------------------------------------+
```
now
```
mysql> SELECT MICROSECONDS_SUB('2010-11-30 23:50:50', 2);
+-------------------------------------------------------------------+
| microseconds_sub(cast('2010-11-30 23:50:50' as DATETIMEV2(0)), 2) |
+-------------------------------------------------------------------+
| 2010-11-30 23:50:49.999998                                        |
+-------------------------------------------------------------------+
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/types/DateTimeV2Type.java [hunk 1]
```diff
@@ -39,6 +39,10 @@ public class DateTimeV2Type extends DateLikeType {
     public static final DateTimeV2Type SYSTEM_DEFAULT = new DateTimeV2Type(0);
     public static final DateTimeV2Type MAX = new DateTimeV2Type(MAX_SCALE);
 
+    public static final DateTimeV2Type MICRO = new DateTimeV2Type(MAX_SCALE);
```
Comment (zclllyybb) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/types/DateTimeV2Type.java:42]: why need this? I think MAX is just enough for our previous practice

**Review by zclllyybb — CHANGES_REQUESTED** [review]
should add many testcases

**Comment by Mryange** [comment]
run buildall

File: regression-test/suites/correctness/test_from_millisecond_microsecond.groovy
Comment (zclllyybb) [inline regression-test/suites/correctness/test_from_millisecond_microsecond.groovy:1]: please also add some testcase from table. and some with scale is not 0 or 6, either

**Comment by Mryange** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/DateTimeArithmetic.java [hunk 2]
```diff
@@ -210,14 +210,6 @@ public static Expression secondsAdd(DateTimeV2Literal date, IntegerLiteral secon
         return date.plusSeconds(second.getValue());
     }
 
-    /**
-     * datetime arithmetic function microseconds-add.
-     */
-    @ExecFunction(name = "microseconds_add", argTypes = {"DATETIMEV2", "INT"}, returnType = "DATETIMEV2")
```
Comment (zclllyybb) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/DateTimeArithmetic.java:216]: restore this and make `DateTimeV2Literal.plusMicroseconds` always use MICRO pricision

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by Mryange** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by zclllyybb** [comment]
run performance

**Review by zhiqiang-hhhh — APPROVED** [review]

**Comment by zhiqiang-hhhh** [comment]
run performance

**Review by yiguolei — APPROVED** [review]

**Pull request merged**